src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,195 @@
+            case 'taxRequired':
+                return translate('violations.taxRequired');
+            case 'smartscanFailed':
+                return translate('violations.smartscanFailed');
+            case 'taxAmountChanged':
+                return translate('violations.taxAmountChanged');
+            case 'receiptRequired':
+                return translate('violations.receiptRequired', {
+                    amount: violation.data?.amount ?? '0',
+                    category: violation.data?.category ?? '',
+                });
+            case 'cashExpenseWithNoReceipt':
+                return translate('violations.cashExpenseWithNoReceipt', {amount: violation.data?.amount ?? ''});
+            case 'categoryOutOfPolicy':
+                return translate('violations.categoryOutOfPolicy');
+            case 'modifiedDate':
+                return translate('violations.modifiedDate');
+            case 'taxRateChanged':
+                return translate('violations.taxRateChanged');
+            case 'customUnitOutOfPolicy':
+                return translate('violations.customUnitOutOfPolicy');
+            case 'overLimit':
+                return translate('violations.overLimit', {amount: violation.data?.amount ?? ''});
+            case 'overLimitAttendee':
+                return translate('violations.overLimitAttendee', {amount: violation.data?.amount ?? ''});
+            case 'receiptNotSmartScanned':
+                return translate('violations.receiptNotSmartScanned');
+            case 'allTagLevelsRequired':
+                return translate('violations.allTagLevelsRequired');
+            case 'nonExpensiworksExpense':
+                return translate('violations.nonExpensiworksExpense');
+            case 'overAutoApprovalLimit':
+                return translate('violations.overAutoApprovalLimit', {formattedLimitAmount: violation.data?.formattedLimitAmount ?? ''});
+            case 'duplicatedTransaction':
+                return translate('violations.duplicatedTransaction');
+            case 'someTagLevelsRequired':
+                return translate('violations.someTagLevelsRequired');
+            case 'overCategoryLimit':
+                return translate('violations.overCategoryLimit', {categoryLimit: violation.data?.categoryLimit ?? ''});
+            case 'perDayLimit':
+                return translate('violations.perDayLimit', {limit: violation.data?.limit ?? ''});
+            default:
+                // The interpreter should never get here because the switch cases should be exhaustive.
+                //
+                // If typescript is showing an error on the assertion below it means the switch statement is out of
+                // sync with the `ViolationNames` type, and one or the other needs to be updated.
+                //
+                // The 'unnecessary' type assertion below is the guarantee of type safety for these translations.
```

Comment:
Why is that? Just curious

Reply:
The types are narrowed by each case.

The typescript compiler traces the control flow of the value and modifies the type in branches where the set of possible values is reduced.

https://www.typescriptlang.org/docs/handbook/2/narrowing.html

Reply:
Here's a quick example:

```
type Foo = "Foo"
type Bar = "Bar"
type FooBar = Foo | Bar

function doFooBarBaz(fooBarBaz: FooBar) {
    if(fooBar === "foo") {
       const foo:Foo = fooBar  // this is ok!
       const bar:Bar = fooBar  // this will have a type error, because fooBar cannot possibly be 'bar'
       return [foo, bar];
   }

   const foo: Foo =  fooBar    // this will throw an error because if `fooBar === 'foo'` we've already returned above
   const bar:Bar = fooBar   // this will be fine
}
```

Typescript is a lot different than other languages in that the compiler follows the flow of logic through the system and sort of evaluates your code without running it to give really helpful errors.